### ISSUE TITLE
Using requires instead of imports prevents from prerendering

### ISSUE THREAD

**Issue by niwolive** [chrisvfritz/prerender-spa-plugin#23]
It seems using require statements (e.g. `const mod = require('module');` instead of imports (e.g. `import mod from 'module' `) anywhere in the code prevents any prerendering from occuring.

However, the code still works properly in the browser.

I certainly need to investigate that more and will follow up with a more precise example if there is interest. Meanwhile I'm curious if any of you can confirm / explain this behaviour.

**Comment by chrisvfritz** [comment]
I haven't encountered this. A minimal example reproducing the problem will definitely help! 🙂

**Comment by chrisvfritz** [comment]
The [2.0 release](https://github.com/chrisvfritz/prerender-spa-plugin/releases/tag/v2.0.0) was just pushed out, so it's also worth testing that the problem still occurs.

**Comment by fuzzytern** [comment]
Sounds great. I'm away from keyboard until mid-january. Will definitely check then.

**Comment by u12206050** [comment]
I might also have this issue even on V2.0, I have the following code where I am using require
```
Vue.component('login', function (resolve) {
  require(['./login.vue'], resolve)
})
```
this is so I can lazy load the login component.

The code works, but prerender doesn't generate any prerendered html on any of my paths, in fact when ever I have another route besides '/' that I want prerender, the build just never stops without any warnings or errors. Have been waiting for more than 30 minutes.

**Comment by chrisvfritz** [comment]
@u12206050 I think you're getting a different problem. We use [core-js](https://github.com/chrisvfritz/prerender-spa-plugin/blob/master/lib/phantom-page-render.js#L13-L15) to polyfill Promise and other features, but in your case, `Promise` is undefined when your script runs. 😕  I'll try to look into this soon.

**Comment by u12206050** [comment]
@chrisvfritz Thanks, have you been able to find out anything? I know that the use of vue-material is currently limiting my chances of getting any sort of prerendering working.

**Comment by toverux** [comment]
I've also had that `ReferenceError: Can't find variable: Promise` error, I've fixed it (before seeing this issue) by adding core-js' Promise polyfill in my app.

```ts
import 'core-js/es6/promise';
```

@chrisvfritz are you saying that I shouldn't do that just to support Phantom?

**Comment by lyquocnam** [comment]
is this fixed ?

**Comment by drewlustro** [comment]
Hello @lyquocnam ! Could you please provide code to reproduce your issue?

**Comment by drewlustro** [comment]
Are you all including a `Promise` polyfill in your own code?

**Comment by alanaasmaa** [comment]
@drewlustro Manually editing this file like this has fixed the same issue for me on 3 PC's.

https://github.com/chrisvfritz/prerender-spa-plugin/pull/54/commits/85d7d8cf722d9358c7e9b97def6b72d2e130e61c

**Comment by drewlustro** [comment]
@alanaasmaa thanks for the info! 👍

**Comment by drewlustro** [comment]
@niwolive @u12206050 @lyquocnam @alanaasmaa – this has been fixed in #54, which was merged in last week and published on `npm`.

Cheers.

**Comment by lizhaohuiGit** [comment]
i have the same problem when i use the newest version。 
![image](https://user-images.githubusercontent.com/7029645/35431158-5fff7538-02b6-11e8-930e-123e25c389ec.png)
But i change the old version code. And it works.
